os/arch/arm/src/armv7-a/arm_addrenv.c

```diff
@@ -543,124 +517,95 @@ ssize_t up_addrenv_heapsize(const group_addrenv_t *addrenv)
  *
  ****************************************************************************/
 
-int up_addrenv_select(const group_addrenv_t *addrenv,
-                      save_addrenv_t *oldenv)
+int up_addrenv_select(const group_addrenv_t *addrenv, save_addrenv_t *oldenv)
 {
-  uintptr_t vaddr;
-  uintptr_t paddr;
-  int i;
-
-  binfo("addrenv=%p oldenv=%p\n", addrenv, oldenv);
-  DEBUGASSERT(addrenv);
-
-  for (vaddr = CONFIG_ARCH_TEXT_VBASE, i = 0;
-       i < ARCH_TEXT_NSECTS;
-       vaddr += SECTION_SIZE, i++)
-    {
-      /* Save the old L1 page table entry */
-
-      if (oldenv)
-        {
-          oldenv->text[i] = mmu_l1_getentry(vaddr);
-        }
-
-      /* Set (or clear) the new page table entry */
-
-      paddr = (uintptr_t)addrenv->text[i];
-      if (paddr)
-        {
-          binfo("text: set l1 entry (paddr=%x vaddr=%x)\n", paddr, vaddr);
-          mmu_l1_setentry(paddr, vaddr, MMU_L1_PGTABFLAGS);
-        }
-      else
-        {
-          binfo("text: clear l1 (vaddr=%x)\n", vaddr);
-          mmu_l1_clrentry(vaddr);
-        }
-    }
-
-  for (vaddr = CONFIG_ARCH_DATA_VBASE, i = 0;
-       i < ARCH_DATA_NSECTS;
-       vaddr += SECTION_SIZE, i++)
-    {
-      /* Save the old L1 page table entry */
-
-      if (oldenv)
-        {
-          oldenv->data[i] = mmu_l1_getentry(vaddr);
-        }
-
-      /* Set (or clear) the new page table entry */
-
-      paddr = (uintptr_t)addrenv->data[i];
-      if (paddr)
-        {
-          binfo("data: set l1 entry (paddr=%x vaddr=%x)\n", paddr, vaddr);
-          mmu_l1_setentry(paddr, vaddr, MMU_L1_PGTABFLAGS);
-        }
-      else
-        {
-          binfo("data: clear l1 (vaddr=%x)\n", vaddr);
-          mmu_l1_clrentry(vaddr);
-        }
-    }
+	uintptr_t vaddr;
+	uintptr_t paddr;
+	int i;
+
+	binfo("addrenv=%p oldenv=%p\n", addrenv, oldenv);
+	DEBUGASSERT(addrenv);
+
+	for (vaddr = CONFIG_ARCH_TEXT_VBASE, i = 0; i < ARCH_TEXT_NSECTS; vaddr += SECTION_SIZE, i++) {
+		/* Save the old L1 page table entry */
+
+		if (oldenv) {
+			oldenv->text[i] = mmu_l1_getentry(vaddr);
+		}
+
+		/* Set (or clear) the new page table entry */
+
+		paddr = (uintptr_t) addrenv->text[i];
```

Comment:
same here

Reply:
I will check all the typecasting issue and change it